workflow/artifacts/oss/oss.go

```diff
@@ -236,12 +236,21 @@ func (ossDriver *ArtifactDriver) ListObjects(artifact *wfv1.Artifact) ([]string,
 			if err != nil {
 				return !isTransientOSSErr(err), err
 			}
-			results, err := bucket.ListObjectsV2(oss.Prefix(artifact.OSS.Key))
-			if err != nil {
-				return !isTransientOSSErr(err), err
-			}
-			for _, object := range results.Objects {
-				files = append(files, object.Key)
+			continueToken := ""
+			for {
+				results, err := bucket.ListObjectsV2(oss.Prefix(artifact.OSS.Key), oss.ContinuationToken(continueToken))
+				if err != nil {
+					return !isTransientOSSErr(err), err
+				}
+				// Add to files. By default, 100 records are returned at a time.
+				for _, object := range results.Objects {
+					files = append(files, object.Key)
+				}
+				continueToken = results.NextContinuationToken
+				if !results.IsTruncated {
+					break
+
+				}
```

Comment:
We should check whether it's truncated before accessing next continuation token

Reply:
Yes.